files/src/hobby-hub

```diff
@@ -0,0 +1,509 @@
+                shift # Remove argument name from processing
+            else
+                TRANSFER_FILE="$2"
+                if [ -f $TRANSFER_FILE ]
+                then
+                    TRANSFER=1
+                else
+                    TRANSFER=2
+                fi
+                shift # Remove argument name from processing
+                shift # Remove argument value from processing
+            fi
+        ;;
+        -z|--remove) # Remove program
+        if [ -z "$2" ]
+            then
+                REMOVE=2
+                echo nope
+                shift # Remove argument name from processing
+            else
+                
+                REMOVE_FILE="$2"
+                if [ -f /etc/hobby-hub/test_programs/$REMOVE_FILE ]
+                then
+                    REMOVE=1
+                else
+                    REMOVE=2
+                fi
+                shift # Remove argument name from processing
+                shift # Remove argument value from processing
+            fi
+        ;;
+        -d|--debug) # Debug provided program
+            if [ -z "$2" ]
+            then
+                DEBUG=2
+                shift # Remove argument name from processing
+            else
+                DEBUG_FILE="$2"
+                if [ -f $DEBUG_FILE ]
+                then
+                    DEBUG=1
+                else
+                    DEBUG=2
+                fi
+                shift # Remove argument name from processing
+                shift # Remove argument value from processing
+            fi
+        ;;
+        -dr|--dbgResume) # resume's from debug state
+        DBGRESUME=1
+        shift
+        ;;
+        *)
+        OTHER_ARGUMENTS+=("$1")
+        HELP=1
+        shift # Remove generic argument from processing
+        ;;
+    esac
+
+    break # don't allow more than one argument. for the time being...
+
+done
+
+#---------------------------------------------------------------------------------------------------------#
+
+
+if [ $HELP -eq 1 ]
+then
+    echo options:
+    echo "[-h|--help]                     usage instructions. Help output."
+    echo "[-r|--restart]                  restarts the currently running programs."
+    echo "[-s|--stop]                     kills the currently running and paused programs. "
+    echo "[-p|--pause]                    pauses the currently running programs."
+    echo "[-c|--continue]                 resumes currently paused programs."
+    echo "[-l|--list]                     lists currently running programs."
+    echo "[-t|--transfer] program_file    transfers program to staging directory."
+    echo "[-z|--remove] program_file      stop and remove provided program."
+    echo "[-d|--debug] program_file       temporarly runs provided program file."
+    echo "[-dr|--dbgResume]               terminates debug and resumes state."
+elif [ -s /etc/hobby-hub/out/debug_config.txt ]
+then
+# we're in a debug session
```

Comment:
How does debugging work? Is there documentation anywhere?

Reply:
debugging will pause all of the currently running programs and just run the program you provide without actually sending it to the staging directory. 

The idea is that if you want to test your code without actually sending it to the program manager and crashing everything.

Reply:
https://github.com/Senior-Design-0x07/the-hobby-hub/wiki/Interfacing-With-The-Program-Manager----Utility